examples/pytorch/rgcn/link.py

```diff
@@ -90,47 +218,55 @@ def main(args):
         loss.backward()
         nn.utils.clip_grad_norm_(model.parameters(), max_norm=1.0) # clip gradients
         optimizer.step()
-
         print("Epoch {:04d} | Loss {:.4f} | Best MRR {:.4f}".format(epoch, loss.item(), best_mrr))
-
         if (epoch + 1) % 500 == 0:
             # perform validation on CPU because full graph is too large
             model = model.cpu()
             model.eval()
-            print("start eval")
             embed = model(test_g, test_nids)
             mrr = calc_mrr(embed, model.w_relation, test_mask, triplets,
-                           batch_size=500, eval_p=args.eval_protocol)
+                           batch_size=500)
             # save best model
             if best_mrr < mrr:
                 best_mrr = mrr
-                th.save({'state_dict': model.state_dict(), 'epoch': epoch}, model_state_file)
-
+                torch.save({'state_dict': model.state_dict(), 'epoch': epoch}, model_state_file)
             model = model.to(device)
 
-    print("Start testing:")
-    # use best model checkpoint
-    checkpoint = th.load(model_state_file)
+if __name__ == '__main__':
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+    print(f'Training with DGL built-in RGCN module')
+
+    # load and preprocess dataset
+    data = FB15k237Dataset(reverse=False)
+    g = data[0]
+    num_nodes = g.num_nodes()
+    num_rels = data.num_rels
+    train_g = get_subset_g(g, g.edata['train_mask'], num_rels)
+    test_g = get_subset_g(g, g.edata['train_mask'], num_rels, bidirected=True)
+    test_g.edata['norm'] = dgl.norm_by_dst(test_g).unsqueeze(-1)
+    test_nids = torch.arange(0, num_nodes)
+    test_mask = g.edata['test_mask']
+    subg_iter = SubgraphIterator(train_g, num_rels, 'uniform')
+    dataloader = GraphDataLoader(subg_iter, batch_size=1, collate_fn=lambda x: x[0])
+
+    # Prepare data for metric computation
+    src, dst = g.edges()
+    triplets = torch.stack([src, g.edata['etype'], dst], dim=1)
+
+    # create RGCN model
+    model = LinkPredict(num_nodes, 500, num_rels).to(device)
```

Comment:
Perhaps it's better to set a default value of `h_dim` in the class definition.

Reply:
Sure.